numba/cpython/setobj.py

```diff
@@ -1323,6 +1364,7 @@ def set_update(context, builder, sig, args):
         # set instance
         casted = context.cast(builder, loop.value, items_type.dtype, inst.dtype)
         inst.add(casted)
+        context.nrt.decref(builder, items_type.dtype, loop.value)
```

Comment:
I think you need the same comment above for_iter as in the previous constructor.